.github/workflows/application-onboarding-or-promotion-automation.yml

```diff
@@ -0,0 +1,95 @@
+name: Automation for application onboarding or promotion
+
+on:
+  workflow_dispatch:
+    inputs:
+      application_name:
+        description: Application name
+        required: true
+        type: string
+      application_environment:
+        description: Application environment
+        required: true
+        type: string
+      maintainer_github_user:
+        description: Maintainer GitHub username
+        required: true
+        type: string
+      task:
+        description: Onboarding or promoting
+        required: true
+        type: string
+  workflow_call:
```

Comment:
Is it weird that we are using both `workflow_call` and `workflow_dispatch`? Wasn't exactly sure about the difference, so I pulled these from the docs:

> Use on.workflow_call to define the inputs and outputs for a reusable workflow

> When using the workflow_dispatch event, you can optionally specify inputs that are passed to the workflow.

Reply:
oh, okay. I see. thanks!